src/react-components/room/MicSetupModal.js

```diff
@@ -0,0 +1,140 @@
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import { Modal } from "../modal/Modal";
+import { Button } from "../input/Button";
+import { ReactComponent as ChevronBackIcon } from "../icons/ChevronBack.svg";
+import { ReactComponent as MicrophoneIcon } from "../icons/Microphone.svg";
+import { ReactComponent as MicrophoneMutedIcon } from "../icons/MicrophoneMuted.svg";
+import { ReactComponent as VolumeHighIcon } from "../icons/VolumeHigh.svg";
+import { ReactComponent as VolumeOffIcon } from "../icons/VolumeOff.svg";
+import styles from "./MicSetupModal.scss";
+import { IconButton } from "../input/IconButton";
+import { SelectInputField } from "../input/SelectInputField";
+import { ToggleInput } from "../input/ToggleInput";
+import { ToolbarButton } from "../input/ToolbarButton";
+
+const micButtonDiameter = 96;
+
+export function MicSetupModal({
+  className,
+  onPromptMicrophone,
+  selectedMicrophone,
+  microphoneOptions,
+  onChangeMicrophone,
+  microphoneEnabled,
+  micLevel,
+  soundPlaying,
+  microphoneMuted,
+  onChangeMicrophoneMuted,
+  onEnterRoom,
+  onBack,
+  ...rest
+}) {
+  return (
+    <Modal
+      title="Microphone Setup"
+      beforeTitle={
+        <IconButton onClick={onBack}>
+          <ChevronBackIcon />
+          <span>Back</span>
+        </IconButton>
+      }
+      className={className}
+      contentClassName={styles.content}
+      {...rest}
+    >
+      <p>Check your microphone and audio before entering.</p>
+      <div className={styles.audioCheckContainer}>
+        <ToolbarButton
+          icon={
+            microphoneEnabled ? (
+              <MicrophoneIcon width={48} height={48} />
+            ) : (
+              <MicrophoneMutedIcon width={48} height={48} />
+            )
+          }
+          label={microphoneEnabled ? "Talk to Test Microphone" : "Microphone Disabled"}
+          className={classNames(styles.largeToolbarButton, styles.micButton)}
+          iconContainerClassName={styles.micButtonContainer}
+          onClick={onPromptMicrophone}
+          disabled={microphoneEnabled}
+          large
+        >
+          <div
+            className={styles.micLevelIcon}
+            style={{
+              clip: `rect(${micButtonDiameter -
+                Math.floor(micLevel * micButtonDiameter)}px, ${micButtonDiameter}px, ${micButtonDiameter}px, 0px)`
+            }}
```

Comment:
This effect is cool but kinda busy visually. Might be nice to use the same visuals for the microphone as it appears later on (and make similar adjustments to the speaker icon rather than have the entire circle glow blue)

![image](https://user-images.githubusercontent.com/4072106/93264815-8940d200-f75c-11ea-8ba3-6a18b26a64fb.png)